### PR TITLE
feat(team-store): Store loading / hasMore state in store

### PR DESCRIPTION
**Pull request by evanpurkhiser**
We'll need this state in the future for the upcoming `useTeams` hook

### CODE REVIEW

File: static/app/stores/teamStore.tsx [hunk 1]
```diff
@@ -49,7 +63,11 @@ const teamStoreConfig: Reflux.StoreDefinition & TeamStoreInterface = {
     const item = this.getBySlug(itemId);
 
     if (!item) {
-      this.state.push(response);
+      this.state = {
+        ...this.state,
+        teams: [...this.state.teams, response],
```
Comment (davidenwang) [inline static/app/stores/teamStore.tsx:68]: not sure if this is pretty typical for all our stores but these names feel a little generic. `response` seems to be an updated team object, but I can't really tell that from the name `response`. It's out of scope for this PR but might be worth updating later

File: static/app/stores/teamStore.tsx [hunk 2]
```diff
@@ -59,44 +77,46 @@ const teamStoreConfig: Reflux.StoreDefinition & TeamStoreInterface = {
     // components use stores correctly. To be safe reload browser :((
     if (response.slug !== itemId) {
       // Remove old team
```
Comment (davidenwang) [inline static/app/stores/teamStore.tsx:79]: can we change this comment to reflect that we are removing the old team and adding the new team

**Review by davidenwang — APPROVED** [review]
looks goood

File: static/app/stores/teamStore.tsx (hunk 2 above)
Comment (EvanPurkhiser) [reply static/app/stores/teamStore.tsx:79]: Yeah good call

Comment (EvanPurkhiser) [reply static/app/stores/teamStore.tsx:79]: Let me do it in a follow up tho

**Pull request merged**